Cell.py

```diff
@@ -4,12 +4,86 @@ def __init__(self, loc):
 		# Parameters:
 		# 	- loc: list of 2 integers
 		# 		tells the location of the cells
-		self.loc 				= loc
+		self.loc 				= loc       # [row, col] 
 		self.isOccupied 		= False
 		self.isTransformable 	= False
-		self.Score 				= 0
+		self.Score 				= 0 		# transformable = 100, 
 		self.isAvailable		= False
 		self.isBlack			= ((loc[0] + loc[1]) % 2) == 0
 
 		self.piece				= "" # chess piece objects
-		
+
+	def setPiece(piece):
+
+
+	def removePiece():
+
+
+	def showPossibleMoves(currentBoardState):
+		"""
+		Parameters:
+			- currentBoardState: Board object
+				shows the current state of the chess board
+		Return:
+			- `listOfPossibleMoves`: list<list<2 integer>>
+				tells the all the possible of a piece ON this CELL
+		Logic:
+			`listOfLegalMoves` is a MUST-HAVE parameter 
+			in the main area of the function, `listOfLegalMoves`
+			will be used to check if it is any conflict like,
+			move on the same cell or eat the same color piece.
+		"""
+		if self.isOccupied == False:
+			return list()
+
+		listOfLegalMoves = list()
+		name = self.piece.name
+
+		if name == "king":
+			listOfLegalMoves = list(
+				set(
```

Comment:
Does `set( )` aim to eliminate repeating data, before converting to `list( )` data type?

Reply:
Yep. that's correct